cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -217,41 +380,42 @@ func printUpgradePlan(up *upgrade.Upgrade, plan *outputapi.UpgradePlan, unstable
 			continue
 		} else if component.Name == constants.Kubelet {
 			if printManualUpgradeHeader {
-				fmt.Fprintln(w, "Components that must be upgraded manually after you have upgraded the control plane with 'kubeadm upgrade apply':")
-				fmt.Fprintln(tabw, "COMPONENT\tCURRENT\tTARGET")
-				fmt.Fprintf(tabw, "%s\t%s\t%s\n", component.Name, component.CurrentVersion, component.NewVersion)
+				printer.Fprintln(writer, "Components that must be upgraded manually after you have upgraded the control plane with 'kubeadm upgrade apply':")
+				plan := newComponentUpgradePlan(component.Name, component.CurrentVersion, component.NewVersion)
+				printer.PrintObj(&plan, writer)
 				printManualUpgradeHeader = false
 			} else {
-				fmt.Fprintf(tabw, "%s\t%s\t%s\n", "", component.CurrentVersion, component.NewVersion)
+				plan := newComponentUpgradePlan("", component.CurrentVersion, component.NewVersion)
+				printer.PrintObj(&plan, writer)
 			}
 		} else {
 			if printHeader {
 				// End of manual upgrades table
-				endOfTable()
-
-				fmt.Fprintf(w, "Upgrade to the latest %s:\n", up.Description)
-				fmt.Fprintln(w, "")
-				fmt.Fprintln(tabw, "COMPONENT\tCURRENT\tTARGET")
+				printer.Flush(writer, false)
+				printer.Fprintln(writer, "")
+				printer.Fprintf(writer, "Upgrade to the latest %s:\n", up.Description)
+				printer.Fprintln(writer, "")
 				printHeader = false
 			}
-			fmt.Fprintf(tabw, "%s\t%s\t%s\n", component.Name, component.CurrentVersion, component.NewVersion)
+			plan := newComponentUpgradePlan(component.Name, component.CurrentVersion, component.NewVersion)
+			printer.PrintObj(&plan, writer)
 		}
 	}
-	// End of control plane table
-	endOfTable()
+	printer.Flush(writer, true)
+	printer.Fprintln(writer, "")
```

Comment:
here a new line is added again.
perhaps Flush functions can have the new line...not sure.

Reply:
This is just for table print. I move the new line into the table flush.